iapetus/porin_membrane_system.py

```diff
@@ -0,0 +1,179 @@
+            atoms_to_freeze = top.select('protein or resname ' + self.ligand)
+        # Perform the minimization of the ligand-porin-membrane
+        self._minimize_energy(atoms_to_freeze, platform, max_iterations)
+
+    def _place_ligand(self, structure, ligand_structure, porin_indices):
+
+        """
+        Place the ligand at the geometrical center of the porin's bottom and
+        get the resulting ligand-porin-membrane ParMed Structure.
+
+        Parameters
+        ----------
+        structure : object
+            The porin-membrane system ParMed Structure
+        ligand-structure : object
+            The ligand ParMed structure
+        porin_indices : list
+            List of indices corresponding to the porin atoms
+
+        Returns
+        ----------
+            The ligand-porin-membrane ParMed Structure
+
+        """
+
+        coords = structure.get_coordinates(frame=0)
+        coords_ligand = ligand_structure.get_coordinates(frame=0)
+        coords_ligand += coords[porin_indices,:].mean(0) - coords_ligand[:,:].mean(0)
+        coords_ligand[:,2] += coords[porin_indices,:].min(0)[2] - coords[porin_indices,:].mean(0)[2]
+        ligand_structure.coordinates = coords_ligand
+        new_structure = structure + ligand_structure
+
+        return new_structure
+
+    def _minimize_energy(self, atoms_to_freeze, platform, max_iterations):
+        """
+        Use the OpenMM minimizeEnergy method in a system with the
+        ligand-porin-membrane atoms frezeed. Only the water molecules and ions
+        will moved to avoid overlaps with the ligand.
+
+        Parameters
+        ----------
+        atoms_to_freeze : list
+            List of atoms that won't move during minimization
+        platform : object
+            Platform used by OpenMM
+        max_iterations : int, optional, default=2000
+            Maximum number of iterations for minimization.
+                If 0, minimization continues until converged.
+
+        """
+
+        backup_system = self.structure.createSystem(nonbondedMethod=app.PME,
+                                 nonbondedCutoff=1*unit.nanometer,
+                                 rigidWater=True,
+                                 flexibleConstraints=True,
+                                 constraints=app.HBonds,
+                                 hydrogenMass=3*unit.amu,
+                                 removeCMMotion=False)
+        self.system =  deepcopy(backup_system)
+        # zero mass = atoms won't move
+        for index in atoms_to_freeze:
+            backup_system.setParticleMass(int(index), 0.0*unit.dalton)
+        integrator = mm.LangevinIntegrator(300*unit.kelvin, 1.0/unit.picoseconds, 2*unit.femtosecond)
+        simulation = app.Simulation(self.structure.topology, backup_system, integrator, platform)
+        simulation.context.setPositions(self.structure.positions)
+        simulation.minimizeEnergy(tolerance=1*unit.kilojoule/unit.mole, maxIterations=max_iterations)
```

Comment:
This might be too tight, or maybe we want to expose `tolerance` as an optional argument with a default.

```python
def _minimize_energy(self, atoms_to_freeze, platform, tolerance=1*unit.kilojoule/unit.mole, max_iterations=2000):
...
```